### ISSUE TITLE
fix: AccountId#toSolidityAddress() does not return expected result when AccountId has a Key Alias

### ISSUE THREAD

**Issue by bguiz** [hashgraph/hedera-sdk-js#1817]
### Problem

```js
    const createAccountTxRecordWithChildren = await new TransactionRecordQuery()
        .setTransactionId(createAccountTxRecord.transactionId)
        .setIncludeChildren(true)
        .execute(client);
    const multisigAccountId =
        createAccountTxRecordWithChildren?.children[0]?.receipt?.accountId;
    const multisigAccountEvmAddress =
        createAccountTxRecordWithChildren?.children[0]?.evmAddress;
    console.log('multisigAccountId', multisigAccountId.toString());
    console.log('multisigAccountEvmAddress', multisigAccountEvmAddress.toString());
    console.log('toLongZeroEvmAddress(multisigAccountId)', toLongZeroEvmAddress(multisigAccountId));
    console.log('multisigAccountId.toSolidityAddress()', multisigAccountId.toSolidityAddress());

```

```text
multisigAccountId 0.0.630405
multisigAccountEvmAddress 6c48bca69c6e7ebcc565f4b79d890eb1fe45e07f
toLongZeroEvmAddress(multisigAccountId) 0000000000000000000000000000000000099e85
multisigAccountId.toSolidityAddress() 0000000000000000000000000000000000099e85

```

Note that this only occurs for accounts when their key alias was originally created with an ECDSA key.
It does not occur when their key alias was originally created with an EdDSA key.

### Solution

* In this scenario I expect `multisigAccountEvmAddress` and `multisigAccountId.toSolidityAddress()` to be the same - the "regular" EVM address
* And expect only `toLongZeroEvmAddress(multisigAccountId)` (which is `EntityIdHelper.toSolidityAddress([shard, realm, num])`) to give me the long zero address

**Comment by petreze** [comment]
@bguiz 
`AccountId.toSolidityAddress()` checks whether or not the `evmAddress` in the `AccountId` is set
- if it is -> it yields the regular EVM address
- if it is not -> it will generate the long-zero format evm address based on the `shard, realm, num` fields

In your example you are getting `multisigAccountId` from the child record which cannot populate the `evmAddress` field, hence it cannot generate the regular evm address because there is no cryptographic relation between the account id and the evm address

**Comment by petreze** [comment]
For your case I will suggest you to do the following:
After getting `const multisigAccountEvmAddress =
        createAccountTxRecordWithChildren?.children[0]?.evmAddress;`
Generate the `AccountId` as follows: `AccountId.fromEvmAddress(0, 0, multisigAccountEvmAddress)` and then cast `toSolidityAddress()`

Here, keep in mind that `Account.fromEvmAddress()` won't populate the `num` field of the `AccountId` because there is no cryptographic or any other link between the evm address and the num field. In order to apply it, you should call `AccountId.populateAccountNum(client)`

**Comment by bguiz** [comment]
Yes, it is possible to get the EVM address in that way, but the issue is that it is unintuitive, and a dev is likely to get stuck after using the wrong address.

> AccountId.toSolidityAddress() checks whether or not the evmAddress in the AccountId is set

Hmm, so if this behaviour needs to remain unchanged, what about adding another method
e.g. `AccountId.getAliasEvmAddress()` which will return the *correct* EVM address.

Where *correct* is here is the address which is usable,
and actually references the intended `AccountId` when passed as a parameter into a HSCS smart contract function.

**Comment by bguiz** [comment]
@petreze The full script is here, including a README with instructions. You'll need a HTS FT with the treasury account ID set to your operator account.
https://github.com/hedera-dev/hedera-code-snippets/tree/feat/multisig-account-sc/multisig-sc-account

**Comment by bguiz** [comment]
Understood - perhaps, as suggested earlier, there should be a new SDK method `AccountId.getAliasEvmAddress()` which performs either a `TransactionReocrd` or a mirror node query to obtain the "correct" address.

(Where "correct" is the address that would work in HTS transactions, or other similar cases)

> This is the reason why the evmAddress field was added to the TransactionRecord of the child TX, so that you can generate AccountId.fromEvmAddress(0, 0, multisigAccountEvmAddress) or alternately query the mirror node, there is no other way.

**Comment by petreze** [comment]
@bguiz 
We discussed the proposal and as there are several methods (`AccountId.fromEvmAddress()/.fromEvmPublicAddress()./fromSolidityAddress()`) that already cause a bit of confusion (despite the last 2 being deprecated), we decided to implement a method querying the mirror node similar to [AccountId.populateAccountNum()](https://github.com/hashgraph/hedera-sdk-js/blob/develop/src/account/AccountId.js#L188-L207). It will get the actual evm address and populate it to the `AccountId` object and returning the object itself. From that point on, the developer can access the `evmAddress` field once populated. In that manner, we will avoid any additional confusions and sort of keep the convention established with `populateAccountNum()`
